lib/liquid/standardfilters.rb

```diff
@@ -86,6 +86,7 @@ def truncatewords(input, words = 15, truncate_string = "...".freeze)
     #   <div class="summary">{{ post | split '//' | first }}</div>
     #
     def split(input, pattern)
+      pattern = pattern.to_s if pattern.is_a?(Numeric)
```

Comment:
@pushrax my first thought was just to `to_s` the pattern, which is what we seem to do with most other filters, however this test https://github.com/Shopify/liquid/blob/master/test/integration/standard_filter_test.rb#L126 will fail then. @dylanahsmith if Liquid doesn't support creating a Regex anyways, maybe this test is bogus, we should just remove it and then use the `to_s` approach?